src/main/java/xyz/nucleoid/plasmid/game/map/template/MapTemplate.java

```diff
@@ -211,13 +265,75 @@ public BlockState get(int x, int y, int z) {
             return this.container.get(x, y, z);
         }
 
+        /**
+         * Adds an entity to this chunk.
+         * <p>
+         * The position of the entity must be relative to the map template.
+         *
+         * @param entity The entity to add.
+         */
+        public void addEntity(Entity entity) {
+            CompoundTag tag = new CompoundTag();
+
+            if (!entity.saveToTag(tag)) { return; }
+
+            // Avoid conflicts.
+            tag.remove("UUID");
+
+            ListTag posTag = new ListTag();
+            BlockPos pos = entity.getBlockPos();
+
+            int offsetX = pos.getX() & 15;
+            int offsetY = pos.getY() & 15;
+            int offsetZ = pos.getZ() & 15;
+
+            posTag.add(DoubleTag.of(entity.getX() - (double) (pos.getX() - offsetX)));
+            posTag.add(DoubleTag.of(entity.getY() - (double) (pos.getY() - offsetY)));
+            posTag.add(DoubleTag.of(entity.getZ() - (double) (pos.getZ() - offsetZ)));
+
+            tag.put("Pos", posTag);
+
+            // AbstractDecorationEntity has special position handling with an attachment position.
+            if (entity instanceof AbstractDecorationEntity) {
+                AbstractDecorationEntity decorationEntity = (AbstractDecorationEntity) entity;
+                pos = decorationEntity.getDecorationBlockPos();
+
+                int blockPosX = pos.getX() & 15;
+                if (offsetX == 0 && blockPosX == 15) { blockPosX = -1; }
```

Comment:
This logic can be simplified by working out the minimum chunk coordinates and subtracting:
```java
int minChunkX = pos.getX() & ~15;
int minChunkZ = pos.getZ() & ~15;
```